File: LeetCode/python/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        stack_size = 0
        for i in range(len(s)):
            if s[i] in [ '(','{','[' ]:
                stack.append(s[i])
                stack_size+=1
                continue
            if s[i]== ']':
                if stack_size>0 and stack[stack_size-1]=='[':
                    stack.pop()
                    stack_size-=1
                    continue
                else:
                    return False
            if s[i]== '}':
                if stack_size>0 and stack[stack_size-1]=='{':
                    stack.pop()
                    stack_size-=1
                    continue
                else:
                    return False
            if s[i]== ')':
                if stack_size>0 and stack[stack_size-1]=='(':
                    stack.pop()
                    stack_size-=1
                    continue
                else:
                    return False
        if stack_size>0:
            return False
        return True

File: LeetCode/JavaScript/ValidParentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    let len = -1;
    for(let i=0 ; i<s.length ; i++){
        if(s[i]=== ')'){
            console.log("  check: )");
            if(len<0)
                return false;
            if(stack[len]!== '(' )
                return false;
            else{
                stack.pop();
                len--;
            }
        }
        else if(s[i]=== '}'){
            if(len<0)
                return false;
            if(stack[len]!== '{' )
                return false;
            else{
                stack.pop();
                len--;
            }
        }
        else if(s[i]=== ']'){
            if(len<0)
                return false;
            if(stack[len]!== '[' )
                return false;
            else{
                stack.pop();
                len--;
            }
        }
        else{
            stack.push(s[i]);
            len++;
        }
    }

    if(len>=0)
        return false;

    return true;
};